### PR TITLE
Feature provider nextjs

### PR DESCRIPTION
**Pull request by abbasnosrati**

### CODE REVIEW

File: components/containers/provider-dashboard/helpers/checkSocialMediaValidation.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export const checkSocialMediaValidation = (
   email: string | null,
   telegram: string | null
 ) => {
-  const emailValidation = email ? validateEmail(email) : true;
+  const emailValidation = email ? validateEmail(email) : false;
```
Comment (alimaktabi) [inline components/containers/provider-dashboard/helpers/checkSocialMediaValidation.ts:42]: use better naming for emailValidation.
the variable is boolean so the name of the variable can be `isEmailVerified`

File: components/containers/provider-dashboard/prize-tap/CardTimer.tsx [hunk 2]
```diff
@@ -90,7 +90,10 @@ export const ProviderDashboardCardTimer = ({
             <p className="prize-card__timer-item-label text-gray90">s</p>
           </div>
         </div>
-      ) : deadline.getTime() - now.getTime() > 0 ? (
+      ) : (deadline.getTime() - now.getTime() > 0 && Number(days) > 0) ||
+        Number(minutes) > 0 ||
+        Number(hours) > 0 ||
+        Number(seconds) > 0 ? (
```
Comment (alimaktabi) [inline components/containers/provider-dashboard/prize-tap/CardTimer.tsx:96]: provide more description of the evalutation by declaring this expression into a variable

File: components/containers/provider-dashboard/prize-tap/OfferPrizeForm/ContactInformation.tsx [hunk 3]
```diff
@@ -50,9 +50,9 @@ const contactFields: ContactField[] = [
   },
   {
     name: "email",
-    placeholder: "example@email.com (Optional)",
+    placeholder: "example@email.com",
     icon: "/assets/images/provider-dashboard/email.svg",
-    require: false,
+    require: true,
```
Comment (alimaktabi) [inline components/containers/provider-dashboard/prize-tap/OfferPrizeForm/ContactInformation.tsx:55]: rename this field to `required`

File: components/containers/provider-dashboard/prize-tap/OfferPrizeForm/PrizeInfo/components/SelectTokenOrNft.tsx [hunk 4]
```diff
@@ -154,8 +154,10 @@ const SelectTokenOrNft = ({ showErrors, isRightChain }: Prop) => {
           <div className="relative">
             <div
               className={`relative border p-5 rounded-2xl ${
-                !insufficientBalance ||
-                Number(data.winnersCount) > 500 ||
+                (!isShowingDetails && !insufficientBalance) ||
+                (Number(data.tokenAmount) &&
+                  Number(data.winnersCount) &&
+                  Number(data.winnersCount) > 500) ||
```
Comment (alimaktabi) [inline components/containers/provider-dashboard/prize-tap/OfferPrizeForm/PrizeInfo/components/SelectTokenOrNft.tsx:160]: provide more description by putting this expression into a variable

File: components/containers/provider-dashboard/prize-tap/OfferPrizeForm/PrizeInfo/components/SelectTokenOrNft.tsx [hunk 5]
```diff
@@ -260,14 +262,14 @@ const SelectTokenOrNft = ({ showErrors, isRightChain }: Prop) => {
...
                 <p className="text-error text-[10px] mt-[2px] m-0 p-0 absolute -bottom-4">
                   The maximum number of winners is 500.
                 </p>
               )}
-            {!insufficientBalance &&
+            {!isShowingDetails &&
+              !insufficientBalance &&
               Number(data.totalAmount) > 0 &&
               Number(data.winnersCount) <= 500 &&
               data.winnersCount &&
```
Comment (alimaktabi) [inline components/containers/provider-dashboard/prize-tap/OfferPrizeForm/PrizeInfo/components/SelectTokenOrNft.tsx:275]: provide more description by putting this expression into a variable

File: components/containers/provider-dashboard/prize-tap/OfferPrizeForm/TimeEnrollLimitation/components/EndDateComp.tsx [hunk 6]
```diff
@@ -41,6 +43,14 @@ const EndDateComp = ({ showErrors }: EndDateCompProp) => {
     setEndDate(e);
   };
 
+  useEffect(() => {
+    const checkList = enrollmentDurations.filter((item) => item.selected);
+    if (checkList.length) {
```
Comment (alimaktabi) [inline components/containers/provider-dashboard/prize-tap/OfferPrizeForm/TimeEnrollLimitation/components/EndDateComp.tsx:48]: use `enrollmentDurations.find` or `enrollmentDurations.findIndex` instead

**Review by alimaktabi — COMMENTED** [review]
dear @abbasnosrati please review and resolve these comments to continue the merge request

thank you for your work

File: components/containers/provider-dashboard/helpers/checkSocialMediaValidation.ts (hunk 1 above)
Comment (abbasnosrati) [reply components/containers/provider-dashboard/helpers/checkSocialMediaValidation.ts:42]: Done!

File: components/containers/provider-dashboard/prize-tap/OfferPrizeForm/ContactInformation.tsx (hunk 3 above)
Comment (abbasnosrati) [reply components/containers/provider-dashboard/prize-tap/OfferPrizeForm/ContactInformation.tsx:55]: Done!

File: components/containers/provider-dashboard/prize-tap/OfferPrizeForm/TimeEnrollLimitation/components/EndDateComp.tsx (hunk 6 above)
Comment (abbasnosrati) [reply components/containers/provider-dashboard/prize-tap/OfferPrizeForm/TimeEnrollLimitation/components/EndDateComp.tsx:48]: Done!

**Pull request merged**